src/Nazara/VulkanRenderer/VulkanSurface.cpp

```diff
@@ -6,6 +6,10 @@
 #include <Nazara/VulkanRenderer/Vulkan.hpp>
 #include <Nazara/VulkanRenderer/Debug.hpp>
 
+#if defined(VK_USE_PLATFORM_XCB_KHR)
+#include <Nazara/Platform/X11/ScopedXCB.hpp>
```

Comment:
This isn't a good idea, as ScopedXCB is an implementation detail from one implementation of Window on Linux (we're planning on adding Wayland support).

Isn't there a way to extract the xcb_connection_t* connection from the xcb_window_t window handle, as I do it on Windows?

Reply:
I don't think that it's possible to get the connection from the xcb_window_t as it is only an ID generated by xcb_generate_id. The resource itself is on the X server and we would still need a connection to ask it.

However, it is possible to wrap the window_handle type into what you want.